File: GoldmanSachs/Q14.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int i = 0, n = nums.size(), res = INT_MAX;
        for (int j = 0; j < n; j++) {
            target -= nums[j];
            while (target <= 0) {
                res = min(res, j - i + 1);
                target += nums[i++];
            }
        }
        if(res==INT_MAX) return 0;
        else return res;
    }
};

File: GoldmanSachs/Q14.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        res = 9999
        i=0
        for j in range(len(nums)):
            target -= nums[j]
            while (target <= 0):
                res = min(res, j - i + 1)
                target += nums[i]
                i += 1
        if(res==9999):
            return 0
        else:
            return res
